test/Ocelot.AcceptanceTests/RoutingWithQueryStringTests.cs

```diff
@@ -312,6 +312,44 @@ public void Should_return_response_200_with_query_string_upstream_template_multi
                 .And(x => _steps.ThenTheResponseBodyShouldBe("Hello from Laura"))
                 .BDDfy();
         }
+
+        [Fact]
+        public void Should_return_response_200_with_query_string_upstream_template_multiple_params_with_same_name_and_map_all_traffic()
+        {
+            var subscriptionId = Guid.NewGuid().ToString();
+            var unitId = Guid.NewGuid().ToString();
+            var port = PortFinder.GetRandomPort();
+
+            var configuration = new FileConfiguration
+            {
+                Routes =
+                [
+                    new()
+                    {
+                        DownstreamPathTemplate = "/api/units/{subscriptionId}/{unitId}/updates",
+                        DownstreamScheme = "http",
+                        DownstreamHostAndPorts =
+                        [
+                            new()
+                            {
+                                Host = "localhost",
+                                Port = port,
+                            },
+                        ],
+                        UpstreamPathTemplate = "/api/subscriptions/{subscriptionId}/updates?unitId={unitId}",

```

Comment:
This is copied and pasted test (user scenario) of old tests.
You should not copy old tests at all!
**You have to show us that your user scenario is fixed by your code improvement!**